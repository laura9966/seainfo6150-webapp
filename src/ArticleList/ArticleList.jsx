import React from "react";
import PropTypes from "prop-types";
import ArticleListItem from "../ArticleListItem/ArticleListItem.jsx";
import styles from "./ArticleList.module.css"

const ArticleList = (props) => {
    let displayContent;
    if (props.articles.length) {
      displayContent = (
        <ul>
          {props.articles.map(article => (
            <li key={article.slug}>
              <ArticleListItem article={article}/>
            </li>
          ))}
        </ul>
      );
    } else {
      displayContent = <div>"You have no data!"</div>
    }

    return (
      <div className = {styles.listStyle}>
          {displayContent}
      </div>
    );
  };

  ArticleList.propTypes = {
    articles: PropTypes.array.isRequired
  };

export default ArticleList;