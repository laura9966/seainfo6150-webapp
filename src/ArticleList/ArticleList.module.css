.container {
    background-color: grey;
    padding: 10px;
    border-radius: 10px;
    margin: 20px;
    box-shadow: red 0px 0px 10px;

}
@supports (display: grid) {
  .container {
    display: grid;
    grid-template-columns: 1fr;
    gap: 10px;
  }

  @media screen and (min-width: 700px) {
    .container {
        gap: 10px;
        grid-template-columns: repeat(2, 1fr);
    }
  }

  @media screen and (min-width: 1000px) {
    .container {
        gap: 20px;
        grid-template-columns: repeat(3, 1fr);
    }
  }
}