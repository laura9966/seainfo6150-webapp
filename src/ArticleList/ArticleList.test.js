
import React, { Component } from 'react';
import ArticleList from "./ArticleList";
import { fireEvent } from "@testing-library/react";
describe("ArticleList tests", () => {
  it("renders correctly", () => {
    const articles1 = [
       {
            title : "A title",
            timeStamp: "2020-02-02",
            displayDate: "2020-02-03",
            slug: "slug1",


        },
        {
            title : "A title",
            timeStamp: "2020-02-02",
            displayDate: "2020-02-03",
            slug: "slug2",

        }
    ]
    const { container } = render(<ArticleList articles={articles1}/>);
    expect(container).toMatchSnapshot();
  });
});
