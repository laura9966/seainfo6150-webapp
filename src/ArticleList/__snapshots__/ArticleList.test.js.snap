// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList tests renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="listStyle"
    >
      <article
        class="box"
      >
        <div
          class="articleImage"
        >
          <img
            alt="The Statue of Liberty's torch heads to new museum"
            class="image"
            src="https://cdn.cnn.com/cnnnext/dam/assets/181122102925-statue-of-liberty-crown-super-169.jpg"
          />
        </div>
        <h2
          class="header"
        >
          The Statue of Liberty's torch heads to new museum
        </h2>
        <button
          class="button"
        >
          show more
        </button>
      </article>
    </div>
  </div>
</div>
`;
